operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -189,7 +192,11 @@ protected void addMetadata(boolean forMatch, R actualResource, final R target, P
     addReferenceHandlingMetadata(target, primary);
   }
 
-  private boolean useSSA(Context<P> context) {
+  protected void sanitizeDesired(R desired, R actual, P primary, Context<P> context) {
+    DesiredResourceSanitizer.sanitizeDesired(desired, actual, primary, context);
```

Comment:
Design-wise, this shouldn't be a static call, but rather a per-type instance so that we use polymorphism instead of having a big if-else cascade based on type with a default no-op being used in most cases.

Reply:
I would not complicate that for now, until this is small, we can refactor when we have more of it.